addons/maintenance/models/maintenance.py

```diff
@@ -334,13 +334,18 @@ def write(self, vals):
         # the stage (stage_id) of the Maintenance Request changes.
         if vals and 'kanban_state' not in vals and 'stage_id' in vals:
             vals['kanban_state'] = 'normal'
-        if vals.get('owner_user_id'):
-            self.message_subscribe_users(user_ids=[vals['owner_user_id']])
         res = super(MaintenanceRequest, self).write(vals)
+        if vals.get('owner_user_id') or vals.get('technician_user_id'):
+            self._add_followers()
         if self.stage_id.done and 'stage_id' in vals:
             self.write({'close_date': fields.Date.today()})
         return res
 
+    def _add_followers(self):
+        for request in self:
+            user_ids = (request.owner_user_id + request.technician_user_id).ids
+            request.message_subscribe_users(user_ids=user_ids)
+
```

Comment:
we need to check to do it in constraint or at least the if condition in the function.